specification/security.md

```diff
@@ -0,0 +1,241 @@
+---
+layout: default
+title: Describing API Security
+parent: The OpenAPI Specification Explained
+nav_order: 8
+---
+
+# Describing API Security
+
+One feature of OpenAPI that can prove useful to API providers and consumers is the means to describe API security. Providing information about the security that protects a given API and its Operations is useful to humans, as they can understand security restrictions and account for them in their implementation, and for tooling that can generate code or provide features that facilitate submitting authorization parameters.
+
+OpenAPI provides the [Security Scheme Object](https://spec.openapis.org/oas/latest.html#security-scheme-object), which contains security definitions that can be referenced either globally or per [Operation](https://spec.openapis.org/oas/v3.1.0#operation-object). Unlike other Component objects, Security Scheme objects are referenced by name using a [Security Requirement Object](https://spec.openapis.org/oas/latest.html#security-requirement-object). A security description for a given API or Operation therefore **must** be defined as a Security Scheme Object as they cannot be declared inline.
+
+<figure style="text-align:center">
+  <object type="image/svg+xml" data="{{site.baseurl}}/img/security-object.svg"></object>
+  <figcaption>Security Scheme objects are referenced as Security Requirements, either <a href="structure.html#minimal-openapi-description-structure">globally</a> or by an <a href="paths.html#the-operation-object">Operation</a>.</figcaption>
+</figure>
+
+A Security Requirement declared for a given Operation takes precedence over global Security Requirements. A number of security mechanisms are supported. Each mechanism is indicated using the `type` property, which is shown in the examples below.
+
+There is currently five supported security types, namely:
+
+- API Keys
+- HTTP Authentication
+- Mutual TLS
+- OAuth 2.0
+- OpenID Connect
+
+Each is discussed in the sections below.
+
+## API Keys
+
+An API key is - generally speaking - the API equivalent of a password and is used to supply a secret known only by a given API consumer and the API provider. There are no formal standards that govern API keys and the industry generally relies on accepted best practices for their definition. OpenAPI therefore provides flexibility in how they are defined.
+
+For example, an API provider might use the HTTP header `api-key` as the means for API consumers to send their API keys with each request. A Security Scheme object can be defined to indicate in which HTTP header the API Key should be sent:
+
+```yaml
+components:
```

Comment:
You might want to just merge this into the example below to show the interaction of the Components Object and the Security Requirement Object clearly.  And to avoid [difficult questions](https://github.com/OAI/oascomply/blob/main/reports/processing-model-details.md#security-requirement-object-example) around how Security Requiremnt Objects are resolved in multi-document OpenAPI Descriptions.

Reply:
Ideally I'd like to keep this example separate for the sake of a) simplicity and b) keeping the flow of the page even. Regards the difficult question, maybe I'll just call that out (very briefly) in the page

Reply:
That makes sense.  I probably would not link to that report as it gets very dense and technical, just note that tools might have inconsistent behavior for resolving Security Requirement Objects in referenced documents.  I suspect most will resolve from the entry document's Components Object but I really don't know for certain.  I think that is what was originally intended.

Reply:
Closing per comments around new text